agent/structs/connect_ca.go

```diff
@@ -102,8 +102,14 @@ type CARoot struct {
 	// active root.
 	RotatedOutAt time.Time `json:"-"`
 
-	// Type of private key used to create the CA cert.
+	// PrivateKeyType is the type of the private key used to sign certificates. It
+	// may be "rsa" or "ec". This is provided as a convenience to avoid parsing
+	// the public key from the certificate.
 	PrivateKeyType string
+
+	// PrivateKeyBits is the length of the private key used to sign certificates.
+	// This is provided as a convenience to avoid parsing the public key from the
```

Comment:
```suggestion
	// This is provided as a convenience to avoid parsing the private key from the
```
Did you mean private?

Reply:
Actually no - the certificate doesn't contain the private key. This is to help us avoid parsing the certificates _public_ key to infer which type of key it's private key is. I can reword that a little to make it clearer.

Reply:
Honestly It's a bit of an over-fitted name, I would have preferred `KeyType` since it makes no sense to have public and private keys with different types anyway but I didn't catch that early enough and it would be a breaking change now to fix it.